Add `alphaMode` and `alphaCutOff` material settings

alexykot opened the pull request:
This adds a way to set `alphaMode` for materials in GLTF. Creating transparent materials with GLTF is impossible without setting `alphaMod=BLEND`.

This seems to be very simple to implement, but I'm not sure what is the long term plan on aligning different interfaces and formats. If this approach is fine - I'll add tests and finalise the PR. If not - let me know what would be better way. I need this feature and happy to implement it in whatever way is best.

---
Updated to full implementation, ready for review/merge.

EliCDavis commented on formats/gltf/model.go:
```diff
@@ -26,6 +26,7 @@ type PolyformModel struct {
 type PolyformMaterial struct {
 	Name                 string
 	Extras               map[string]any
+	AlphaMode            MaterialAlphaMode
```
Let's go ahead and make this a pointer
```suggestion
	AlphaMode            *MaterialAlphaMode
```

EliCDavis commented on formats/gltf/writer.go:
```diff
@@ -371,6 +371,7 @@ func (w *Writer) AddMaterial(mat PolyformMaterial) *int {
 				Extensions: extensions,
 			},
 		},
+		AlphaMode:            &mat.AlphaMode,
```
```suggestion
		AlphaMode:            mat.AlphaMode,
```

EliCDavis commented:
Good catch! This fix is fine, and I am happy to merge. Make sure this is the only change you need to make for your own purposes. I've been able to achieve transparency in three.js and babylon renderers while this bug has existed.

https://github.com/EliCDavis/polyform/blob/main/examples/ufo/main.go#L347-L380

![image](https://github.com/user-attachments/assets/0e5cbb4f-0e72-429d-8465-cd3a6f19102a)

alexykot commented:
Cool, I've switched to pointers and added `alphaCutOff` as requested. Also added relevant writer tests.

Yes indeed, the material settings of the UFO model work in Babylon and Three.js. I copied over the parts of GLTF JSON and it did render transparency. I'm not sure why tho, if I read [the spec](https://github.com/KhronosGroup/glTF/blob/0890b76c62cc762ce82d4010df9bfebb1634839b/specification/2.0/schema/material.schema.json#L43) correctly - it should not work, as default is `OPAQUE` even if `alphaMode` is not set. 

Btw, I also tested against [3dviewer.net](https://3dviewer.net/) and it did *not* work without `BLEND`, so there is clearly a disagreement between renderer implementations. 

Anyway, it's part of the spec, so should exist, and now it does 🙂

EliCDavis commented on formats/gltf/writer.go:
```diff
@@ -369,6 +369,11 @@ func (w *Writer) AddMaterial(mat PolyformMaterial) *int {
 		emissiveFactor = &factor
 	}
 
+	var alphaCutOff *float64
+	if mat.AlphaMode != nil {
+		alphaCutOff = mat.AlphaCutoff // alphaCutOff should only be set if the alphaMode is set
+	}
```
I wanted your thoughts on this implementation. I could see users getting confused by us silently not setting `alphaCutOff` even though they included it. Given it's illegal, it might warrant a panic.

```suggestion
	alphaCutOff := mat.AlphaCutoff
	if mat.AlphaMode == nil && mat.AlphaCutoff != nil {
	        // https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_material_alphacutoff
	        panic("When alphaMode is not defined, alphaCutOff MUST NOT be defined")
	}
```

An alternative approach to this would be preventing the user from ever making an illegal move through the introduction of a new field to the PolyformMaterial that encapsulates the alpha logic.

```golang
type PolyformMaterialAlpha interface {
    Set(mat PbrMetallicRoughness)
}

type PolyformMaterialAlphaBlend struct {
     CutOff *float64
}

func (pmab PolyformMaterialAlphaBlend) Set(mat PbrMetallicRoughness) {
    mat.AlphaMode = MaterialAlphaMode_BLEND
    mat.AlphaCutoff = pmab.CutOff
}
```

Appreciate your close attention to the spec. I totally missed that bit of logic.

alexykot replied:
I see your point, but are you sure this is the best way to solve it?

I think, if we treat it as an invalid input - it warrants an error to be bubbled up and returned from the `WriteText/WriteBinary` top level functions. Inducing a `panic` crash seems a bit harsh reaction for an invalid user input 😃

Also, your suggestion will prevent user from setting wrong parameters, but it also makes the interface quite less ergonomic and implementation - more complex.

I think we should strike a balance here between being careless on one hand and overbearing on the other.  

My first implementation was too careless, let me try adding an error return. Let me know what you think.

alexykot replied:
Made an update, let me know what you think of this version.

EliCDavis replied:
I think this is a good compromise! Thanks again for this fix.
